openwillis/measures/text/speech_attribute.py

```diff
@@ -192,41 +171,22 @@ def filter_whisper(json_conf, measures, speaker_label=None):
     item_data = json_conf["segments"]
 
     if speaker_label is not None:
-        # filter out segments that do not have speaker labels
-        item_data = [
-            segment for segment in item_data if "speaker" in segment
-        ]
-
-    # make a dictionary to map old indices to new indices
+        item_data = [segment for segment in item_data if "speaker" in segment]
+        
     item_data = cutil.create_index_column(item_data, measures)
-
-    # phrase-split
-    phrases_idxs, phrases = cutil.filter_phrases(
-        item_data, speaker_label, measures
-    )
-
-    # turn-split
     if speaker_label is not None:
-        turns_idxs, turns = cutil.filter_turns(
-            item_data, speaker_label, measures
-        )
+        turns_idxs, turns = cutil.filter_turns(item_data, speaker_label, measures, min_turn_length)
+        
     else:
         turns_idxs, turns = [], []
-
-
+    
     # filter json to only include items with start_time and end_time
     filter_json = cutil.filter_json_transcribe(item_data, speaker_label, measures)
-
-    # extract words
-    words = [w["word"] for w in filter_json]
-
-    # entire transcript - by joining all the phrases
-    text = " ".join(phrases)
-
-    text_list = [words, phrases, turns, text]
-    text_indices = [phrases_idxs, turns_idxs]
-
-    return filter_json, text_list, text_indices
+    words = [value["word"] for value in filter_json]
+    text = " ".join(turns)
```

Comment:
This should be in the if statement:

i.e. only run if speaker_label is not None